packages/typegpu/src/tgsl/wgslGenerator.ts

```diff
@@ -148,6 +148,19 @@ export function generateExpression(
     const rhsStr = ctx.resolve(convRhs.value);
     const type = operatorToType(convLhs.dataType, op, convRhs.dataType);
 
+    if (
+      op === '/' && !['f32', 'f16'].includes(convLhs.dataType.type) &&
+      !['f32', 'f16'].includes(convRhs.dataType.type)
```

Comment:
A tiny suggestion. It's a bit more verbose, but since we want to minimize the performance impact of resolution, we can reduce the array instantiations and function invocations here.
```suggestion
      op === '/' &&
      convLhs.dataType.type !== 'f32' &&
      convLhs.dataType.type !== 'f16' &&
      convRhs.dataType.type !== 'f32' &&
      convRhs.dataType.type !== 'f16'
```